Modules/ThirdParty/GDCM/src/gdcm/Source/Common/gdcmCAPICryptographicMessageSyntax.cxx

```diff
@@ -192,7 +192,11 @@ bool CAPICryptographicMessageSyntax::Decrypt(char *output, size_t &outlen, const
   BYTE* cek = NULL;
   HCRYPTMSG hMsg = NULL;
   PCMSG_CMS_RECIPIENT_INFO recipientInfo = NULL;
+  DWORD dwMessageType, cbMessageTypeLen = sizeof(DWORD);
```

Comment:
Was this really necessary to move these to the top?

This change goes against the practice of scope localization where variable should be declared ONLY where they are needed.  

If this really is needed, please also make a PR to the upstream GDCM library.

Reply:
You can find more information in the linked [issue](https://github.com/InsightSoftwareConsortium/ITK/issues/235). Moving declarations to the beginning was simple, seemed to follow a pattern and it solved the problem.

Getting rid of the `goto` statements would require more refactoring. There is more than 4000 `goto` statements in the ThirdParty directory, and 76 of them spread over 11 files in GDCM.